var/spack/repos/builtin/packages/py-scikit-image/package.py

```diff
@@ -29,3 +30,7 @@ class PyScikitImage(PythonPackage):
     depends_on('py-cloudpickle', type=('build', 'run'), when='@0.14:')
     depends_on('py-setuptools', type='build')
     depends_on('py-cython@0.23.4:', type='build')
+
+    conflicts('python@:3.5', when='@0.16.1:')
+    conflicts('py-numpy@:1.15.1', when='@0.16.1:')
+    conflicts('py-scipy@:1.0.1',  when='@0.16.1:')
```

Comment:
You should replace these with `depends_on` so that instead of the concretizer failing, it will choose the right thing to do.

Reply:
Since there is no (explicit) `depends_on(python)` for a `PythonPackage`, The conflict for `python@:3.6` for would still remain though, correct ?